Platform/obc/hal/comms_hal.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file comms_hal.c
+ * @author Arash Yazdani
+ * @date 2020-10-01
+ */
+
+/*
+ * When TRX connected, the stubbed blocks can be used for TRX = off situation.
+ */
+
+
+#include "obc_hal.h"
+#include "comms_hal.h"
+#include "hal.h"
+#include "services.h"
+#include "queue.h"
+
+//For storing the set data
+static Sband_config S_config_reg;
+static Sband_Full_Status S_FS;
+STX_return status;
+
+STX_return HAL_S_getFreq (float * S_freq) {
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getFrequency(&S_config_reg.freq);
+  #else
+    //*S_freq = (float)2210.5;
+    status = IS_STUBBED;
+  #endif
+    *S_freq = S_config_reg.freq;
+    return status;
+};
+
+STX_return HAL_S_getControl (Sband_PowerAmplifier *S_PA) {
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getControl(&S_config_reg.PA.status , &S_config_reg.PA.mode);
+  #else
+    //S_PA->status = 1;
+    //S_PA->mode = 3;
+    status = IS_STUBBED;
+  #endif
+    *S_PA = S_config_reg.PA;
+    return status;
+};
+
+STX_return HAL_S_getEncoder (Sband_Encoder *S_Enc){
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getEncoder(&S_config_reg.enc.scrambler, &S_config_reg.enc.filter, &S_config_reg.enc.modulation, &S_config_reg.enc.rate);
+  #else
+    //S_Enc->scrambler = 1;
+    //S_Enc->filter = 6;
+    //S_Enc->modulation = 5;
+    //S_Enc->rate = 4;
+    status = IS_STUBBED;
+  #endif
+    *S_Enc = S_config_reg.enc;
+    return status;
+}
+
+STX_return HAL_S_getPAPower (uint8_t * S_PA_Power) {
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getPaPower(&S_config_reg.PA_Power);
+  #else
+    //*S_PA_Power = 2;
+    status = IS_STUBBED;
+  #endif
+    *S_PA_Power = S_config_reg.PA_Power;
+    return status;
+};
+
+
+STX_return HAL_S_getStatus (Sband_Status *S_status){
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getStatus(&S_FS.status.PWRGD, &S_FS.status.PWRGD);
```

Comment:
Shouldn't one of these be .TXL?

Reply:
Oops. Yeah, sorry. I'll double-check the other ones, too. Thanks for reviewing the code, Thomas.